src/index.js

```diff
@@ -3,7 +3,21 @@ const core = require('@actions/core')
 
 const { runAction } = require('./release-notify-action')
 
+const actionRef = process.env.GITHUB_ACTION_REF
+
 async function run() {
+  if (actionRef === 'main' || actionRef === 'master') {
+    core.warning(
+      `nearform/github-action-notify-release is pinned at HEAD. We strongly ` +
+        `advise against pinning to "@master" as it may be unstable. Please ` +
+        `update your GitHub Action YAML from:\n\n` +
+        `    uses: 'nearform/github-action-notify-release@master'\n\n` +
+        `to:\n\n` +
+        `    uses: 'nearform/github-action-notify-release@<release/tag version>'\n\n` +
+        `Alternatively, you can pin to any git tag or git SHA in the ` +
+        `repository.`
+    )
+  }
```

Comment:
can we take it one step further and since we have to do the same across different actions set up a repo containing these utilities perhaps? similar to what github is doing with their toolkit https://github.com/actions/toolkit/

Reply:
this is a question :) before doing it, let's discuss!

Reply:
yes, It's not very clear to me either. Let's discuss this.